### ISSUE TITLE
'instanceof' narrowing works awkwardly with empty object types & primitives

### ISSUE THREAD

**Issue by DanielRosenwasser** [Microsoft/TypeScript#14426]
Try [in the playground](https://www.typescriptlang.org/play/#src=class%20Dog%20%7B%0D%0A%7D%0D%0A%0D%0Adeclare%20let%20x%3A%20Dog%20%7C%20string%3B%0D%0A%0D%0Aif%20(x%20instanceof%20Dog)%20%7B%0D%0A%20%20%20%20x%0D%0A%7D%0D%0Aelse%20%7B%0D%0A%20%20%20%20x.toLowerCase()%0D%0A%7D).

```ts
class Dog {
}

declare let x: Dog | string;

if (x instanceof Dog) {
    x
}
else {
    x.toLowerCase()
}
```

Expected: No error, first branch's `x` has type `Dog`, second branch's `x` has type `string`
Actual: Property 'toLowerCase' does not exist on type 'never'. (first branch is `string | Dog`, second is `never`)

We should be accounting for primitives a little less sloppily. `instanceof` checks don't work on primitive types.

**Comment by DanielRosenwasser** [comment]
Specifically, any changes here should ensure that

```ts
function foo(x: number | string) {
    if (x instanceof Number) {
        x
    }
}
```

`x` is not narrowed in the above example.

**Comment by DanielRosenwasser** [comment]
Actually, we should just issue an error when no constituent of a union is a primitive.
